app/src/main/java/com/kickstarter/viewmodels/CommentsViewModel.kt

```diff
@@ -123,6 +134,48 @@ interface CommentsViewModel {
                         commentsList.onNext(it.comments)
                     }
                 }
+
+            this.currentUser.loggedInUser()
+                .compose(Transformers.takePairWhen(this.postComment))
+                .compose(bindToLifecycle())
+                .subscribe {
+                    val comment = Comment.builder()
+                        .body(it.second.first)
+                        .parentId(-1)
+                        .authorBadges(listOf())
+                        .createdAt(it.second.second)
+                        .cursor("")
+                        .deleted(false)
+                        .id(-1)
+                        .repliesCount(0)
+                        .author(it.first)
+                        .build()
+
+                    this.insertComment.onNext(comment)
+                }
+
+            initialProject
+                .compose(Transformers.takePairWhen(this.postComment))
+                .compose(bindToLifecycle())
```

Comment:
could you do here 
`.map { it.first?.let { project -> PostCommentData(
                                project = project,
                                body = it.second.first,
                                clientMutationId = null,
                                parentId = null
                            )}`

Then on the switchMap block you already you can call 
`this.apolloClient.createComment(it)`

Reply:
done